learn/user-guide/why-ballerina/concurrent.md

```diff
@@ -5,48 +5,134 @@ description: Concurrency in Ballerina is enabled by strands, which are lightweig
 keywords: ballerina, ballerina platform, api documentation, testing, ide, ballerina central
 permalink: /learn/user-guide/why-ballerina/concurrent/
 active: concurrent
-intro: Concurrency in Ballerina is enabled by strands, which are lightweight threads. 
+intro: Ballerina's concurrency model supports both threads and coroutines.
 redirct_from:
   - /learn/user-guide/why-ballerina/concurrent
 ---
 
-A single operating system thread can contain multiple strands. A single strand is run at a time in a thread and the strands belonging to a single thread are cooperatively multitasked. Strands allow you to optimize the usage of CPU time and is beneficial in implementing non-blocking I/O operations.
+Ballerina's concurrency model supports both threads and coroutines, and has been designed to have a close correspondence with sequence diagrams.
 
-A worker represents a single strand execution in a function. Concurrent operations in a function are defined by multiple workers as shown in the example below.
+A Ballerina program is executed on one or more threads. A thread can consist of one or more strands, which are language-managed logical threads of control. Only one strand of a particular thread can run at a time. The strands belonging to a single thread are cooperatively multitasked. Therefore, strands of a single thread behave as coroutines relative to each other.
+
+Strands enable cooperative multitasking by "yielding". The runtime scheduler may switch the execution of a strand only when it yields via a wait action, a worker receive action, a library function resulting in a blocking call, etc.
```

Comment:
```suggestion
Strands enable cooperative multitasking by "yielding". The runtime scheduler may switch the execution of a strand only when it yields via a `wait` action, a `worker-receive` action, a library function resulting in a blocking call, etc.
```

Reply:
What is our criteria for emphasizing using backticks? Wait is a keyword but worker-receive is not.

Reply:
Only keywords should be within backticks.

Reply:
worker-receive isn't a keyword, so will change this to

```suggestion
Strands enable cooperative multitasking by "yielding". The runtime scheduler may switch the execution of a strand only when it yields via a `wait` action, a worker receive action, a library function resulting in a blocking call, etc.
```

Reply:
+1